bin/run_genasis.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# 
+#  run_nekbone.sh: 
+#
+# --- Start standard header to set build environment variables ----
+function getdname(){
+   local __DIRN=`dirname "$1"`
+   if [ "$__DIRN" = "." ] ; then
+      __DIRN=$PWD;
+   else
+      if [ ${__DIRN:0:1} != "/" ] ; then
+         if [ ${__DIRN:0:2} == ".." ] ; then
+               __DIRN=`dirname $PWD`/${__DIRN:3}
+         else
+            if [ ${__DIRN:0:1} = "." ] ; then
+               __DIRN=$PWD/${__DIRN:2}
+            else
+               __DIRN=$PWD/$__DIRN
+            fi
+         fi
+      fi
+   fi
+   echo $__DIRN
+}
+thisdir=$(getdname $0)
+[ ! -L "$0" ] || thisdir=$(getdname `readlink "$0"`)
```

Comment:
Lines 5-26 can be replaced with:
realpath=`realpath $0`
thisdir=`dirname $realpath`

Reply:
there are 48 more files with this pattern.
RSCE
Redundant Shell Code Elimination